frontend/javascripts/dashboard/dataset/dataset_settings_view.tsx

```diff
@@ -367,14 +366,14 @@ class DatasetSettingsView extends React.PureComponent<PropsWithFormAndRouter, St
     const dataSource = JSON.parse(formValues.dataSourceJson);
 
     if (dataset != null && this.didDatasourceChange(dataSource)) {
-      await updateDatasetDatasource(this.props.datasetId.name, dataset.dataStore.url, dataSource);
+      await updateDatasetDatasource(dataset.directoryName, dataset.dataStore.url, dataSource);
```

Comment:
@daniel-wer I added this warning as it was kinda unexpected to me that the backend takes care of ignoring updates to the `datasource.id` field. Although this is how it has always been, do you think I should keep this info message or is this simply too much info for the user :)

@fm3 of cause I also value your opinion :)